addons/website/models/ir_ui_view.py

```diff
@@ -138,69 +163,64 @@ def render(self, cr, uid, id_or_xml_id, values=None, engine='ir.qweb', context=N
                 qcontext['main_object'] = view_obj
 
             values = qcontext
+        return super(View, self).render(cr, uid, id_or_xml_id, values=values, engine=engine, context=context)
 
-        return super(view, self).render(cr, uid, id_or_xml_id, values=values, engine=engine, context=context)
-
-    def _prepare_qcontext(self, cr, uid, context=None):
-        if not context:
-            context = {}
-
-        company = self.pool['res.company'].browse(cr, SUPERUSER_ID, request.website.company_id.id, context=context)
+    @api.model
+    def _prepare_qcontext(self):
+        local_context = dict(self.env.context)
+        company = self.env['res.company'].sudo().browse(request.website.company_id.id)
 
         editable = request.website.is_publisher()
-        translatable = editable and context.get('lang') != request.website.default_lang_code
+        translatable = editable and local_context.get('lang') != request.website.default_lang_code
         editable = not translatable and editable
 
         qcontext = dict(
-            context.copy(),
+            local_context,
             website=request.website,
             url_for=website.url_for,
             slug=website.slug,
             res_company=company,
-            user_id=self.pool.get("res.users").browse(cr, uid, uid),
+            user_id=self.env.user,
```

Comment:
To be check : the `user` here is browse has sudo, but in the initial code, we should browse it as current user.